service/storage_service.cc

```diff
@@ -5183,7 +5183,12 @@ future<> storage_service::move_tablet(table_id table, dht::token token, locator:
         auto guard = co_await _group0->client().start_operation(&_abort_source);
 
         while (_topology_state_machine._topology.is_busy()) {
-            rtlogger.debug("move_tablet(): topology state machine is busy");
+            using enum topology::transition_state;
+            const auto tstate = *_topology_state_machine._topology.tstate;
+            if (tstate == tablet_migration || tstate == tablet_migration) {
+                break;
```

Comment:
Now you need to check later if the particular tablet is transitioning and refuse to move it.

Reply:
if (tmap.get_transition_info()) goto fail;

Reply:
@tgrabiec hi Tomek, thank you! now a `runtime_error` is throw if `tmap.get_tablet_transition_info(tid)` returns something. does this make more sense now?

Reply:
> does this make more sense now?


Yes.